internal/util/password/scramsha1.go

```diff
@@ -28,11 +28,19 @@ import (
 
 // SCRAMSHA1Hash computes SCRAM-SHA-1 credentials and returns the document that should be stored.
 func SCRAMSHA1Hash(username, password string) (*types.Document, error) {
-	salt := make([]byte, fixedScramSHA1Params.saltLen)
+	salt := make([]byte, fixedScramSHA1Params.saltLen-4) // minus 4 is needed here to encode to 24 bytes.
```

Comment:
There are no 24 bytes.

There is also this thing https://github.com/FerretDB/FerretDB/blob/b6fd73b429749ed35ad8573cbd24719d32c85538/CONTRIBUTING.md?plain=1#L272-L275